src/jmh/java/com/lmax/disruptor/SimpleBenchmark.java

```diff
@@ -0,0 +1,80 @@
+package com.lmax.disruptor;
+
+import com.lmax.disruptor.dsl.Disruptor;
+import com.lmax.disruptor.dsl.ProducerType;
+import com.lmax.disruptor.util.Constants;
+import com.lmax.disruptor.util.DaemonThreadFactory;
+import com.lmax.disruptor.util.SimpleEvent;
+import com.lmax.disruptor.util.SimpleEventHandler;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OperationsPerInvocation;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+import java.util.concurrent.TimeUnit;
+
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@State(Scope.Thread)
+@Fork(1)
+public class SimpleBenchmark
+{
+    private RingBuffer<SimpleEvent> ringBuffer;
+    private SimpleEventHandler eventHandler;
+    private Disruptor<SimpleEvent> disruptor;
+
+    @Setup
+    public void setup() {
+        disruptor = new Disruptor<>(
+                SimpleEvent::new,
+                Constants.RINGBUFFER_SIZE,
+                DaemonThreadFactory.INSTANCE,
+                ProducerType.SINGLE,
+                new BusySpinWaitStrategy());
+
+        eventHandler = new SimpleEventHandler();
+        disruptor.handleEventsWith(eventHandler);
+
+        ringBuffer = disruptor.start();
+    }
+
+    @Benchmark
+    @OperationsPerInvocation(Constants.ITERATIONS)
+    public void publishSimpleEvents() {
+        for (int i = 0; i < Constants.ITERATIONS; i++) {
+            long sequence = ringBuffer.next();
+            SimpleEvent simpleEvent = ringBuffer.get(sequence);
+            simpleEvent.setValue(i);
+            ringBuffer.publish(sequence);
+        }
+
+        while (eventHandler.lastSeenSequence % Constants.ITERATIONS != Constants.ITERATIONS - 1)
```

Comment:
I can't work out why we need the modulo nonsense here - iterations < ring buffer size, can just wait for an exact match?

Reply:
JMH appears to call `publishSimpleEvents` repeatedly, and I only make the ringbuffer once in setup. So the first iteration it works, second iteration it failed because the `lastSeenSequence` was 2x `Constants.ITERATIONS`

I did have a go at putting an `AtomicLong` in the benchmark which the event handler could increment and the benchmark could check and reset each iteration. Just using the sequence and a modulo seemed like the faster way and kept the benchmark tight?

Reply:
Oh - interesting - so we don't necessarily get a new RingBuffer/handler per invoc.

With my pedant hat on (ok, fine, my default hat) - isn't that condition also true before we do the publish? Should we require the handler's sequence to have increased in addition?

We also appear to be reading a plain old non-volatile long from SimpleEventHandler which is being written to from another thread. I can see why we don't want to pay the cost of updating a volatile long, and yes, the benchmark does terminate, but still, I am unkeen. Might be worth pairing on an alternative here, unless I am in possession of an inappropriate stick end?